src/smpi/internals/smpi_replay.cpp

```diff
@@ -31,22 +37,74 @@ char* sendbuffer=nullptr;
 static int recvbuffer_size=0;
 char* recvbuffer=nullptr;
 
-static void log_timed_action (const char *const *action, double clock){
-  if (XBT_LOG_ISENABLED(smpi_replay, xbt_log_priority_verbose)){
-    char *name = xbt_str_join_array(action, " ");
-    XBT_VERB("%s %f", name, smpi_process()->simulated_elapsed()-clock);
-    xbt_free(name);
-  }
+
+/******************************************************************************
+ * This code manages the request dictionaries which are used by the
+ * assynchronous comunication actions (Isend, Irecv, wait, test).
+ *****************************************************************************/
+
+static std::string build_request_key(int src, int dst, int tag){
+  std::stringstream key_stream;
+  key_stream << src << dst << tag;
```

Comment:
Here's another problem: If we really need to generate this key, we need to distinguish between messages. Such as:

1111 (src = 1, dst = 11, tag = 1) and the message
1111 (src = 1, dst = 1, tag = 11)